### DISCUSSION TITLE
Clean exit in case of code failure

### DISCUSSION THREAD

**Discussion by mehdifirouz** [pyinstaller/pyinstaller#8304] [PyInstaller] [answered]
An executable file created by pyinstaller is run on a windows server using automation software. In case of error, the windows error windows pops up and task remains the in the task manager until someone manually acknowledge the error massage window. I am looking for a way to close the exe file and remove if form task manager in case of error without manual intervention

**Comment by rokm** [comment, marked as answer]
Are you asking about PyInstaller's Unhandled exception dialog, or actual Windows error dialog?

**Reply by mehdifirouz** [reply]
I ran 'myexecutable.exe' in command prompt.  When the code fails, a cmd window will pop up with actual python error message in it. It needs to be acknowledged in order to close and remove the process from task manager.

**Reply by rokm** [reply]
If it is a dialog with python traceback, then this is PyInstaller's Unhandled exception dialog. And it is shown because you are not handling exceptions in your code. So if you don't want to see it, you should implement proper top-level exception handling - for example, wrap your whole entry-point script into a try/catch block, and have the exception handler exit the program.

Or, if your program is meant to be ran from command prompt, then build in console mode instead of windowed/noconsole one.
